Model Registry integration tests

mparkhe opened the pull request:
## What changes are proposed in this pull request?

1. Added integration tests for all Model Registry APIs

2. Backend fixes in `SqlAlchemyStore`:
 - `last_updated_time` for registered model on `UpdateRegisteredModel`, `CreateModelVersion`, `DeleteModelVersion`, `UpdateModelVersion` (stage only)
 - `last_updated_time` from model versions on `UpdateModelVersion`
 - `CreateModelVersion` after `DeleteModelVersion` does not reuse version number in any case

## How is this patch tested?

Added new integration test suite

## Release Notes

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s) does this PR affect?

- [ ] UI
- [ ] CLI
- [x] API
- [ ] REST-API
- [ ] Examples
- [ ] Docs
- [ ] Tracking
- [ ] Projects
- [ ] Artifacts
- [x] Models
- [ ] Scoring
- [ ] Serving
- [ ] R
- [ ] Java
- [ ] Python

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [x] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

Commit:
Integration tests:
 - CreateRegisteredModel
 - UpdateRegisteredModel

Fix:
 - SqlAlchemyStore: last_update_time for registered model on UpdateRegisteredModel

Commit:
- DeleteRegisteredModel
- GetRegisteredModelDetails
- ListRegisteredModels

Commit:
fix time

Commit:
adding integration tests for model version APIs. fixing broken functionality in sql

Commit:
registered model's last_updated_time changes when a new model version is created, an existing model version deleted or changes stage (updated model version).

sueann commented on tests/tracking/test_model_registry.py:
```diff
@@ -0,0 +1,454 @@
+    assert "path/to/model" == mlflow_client.get_model_version_download_uri(name, 1)
+
+
+def test_update_create_model_version_flow(mlflow_client, backend_store_uri):
```
nit: `test_update_model_version_flow` ?

sueann commented on tests/tracking/test_model_registry.py:
```diff
@@ -0,0 +1,454 @@
+    mv3 = mlflow_client.create_model_version(name, "a/b/c", "run_id_2")
+    assert mv3.version == 3
+    assert mv3.registered_model.name == name
+    model_versions_detailed = [rm.latest_versions
```
do we have any tests explicitly calling `client.get_latest_versions` ?

mparkhe replied:
added

sueann reviewed (commented):
looks mostly good to me - a few nits & questions inline. Thanks!

Commit:
code fixes:
 server handler for GetLatestVersions passes through "stages" argument

tests:
 - handler test for GetLatestVersions
 - integration test for GetLatestVersions
 - refactor integration tests

sueann commented on tests/tracking/test_model_registry.py:
```diff
@@ -386,6 +299,40 @@ def test_update_create_model_version_flow(mlflow_client, backend_store_uri):
+            mlflow_client.update_model_version(name, version, stage=stage)
+        mvd = mlflow_client.get_model_version_details(name, version)
+        assert mvd.current_stage == stage
+        version_stage_mapping[version] = mvd
```
do you need this? seems like we are unnecessarily modifying the map in flight?

mparkhe replied:
you are right! Not needed anymore...

sueann reviewed (approved):
lgtm - thanks for doing this! just one comment inline.

Commit:
simply test

Commit:
python 2.7 expects tests.tracking to be a python package.

Commit:
lint + python 2.7

Commit:
de-linting tests.tracking

Commit:
backout wrong change

mparkhe commented on tests/__init__.py:
```diff
@@ -1,3 +1,3 @@
-from mlflow.utils.logging_utils import _configure_mlflow_loggers
+#from mlflow.utils.logging_utils import _configure_mlflow_loggers
```
backed out this accidental change change in `7018d2473aacb95290028bf0f7546c706c0acb5a`

Commit:
Merge branch 'master' of github.com:mlflow/mlflow into registry-integration

Commit:
fixing lint and broken tests

sueann reviewed (approved):
thank you for making tests/tracking lint-compatible!! i see that it needed some clean up as well, so it seems like a good idea anyway.
